### ISSUE TITLE
%20 space at end of Internet Archive ID is kept in "Read" url

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#467]
When an IA ID is added with a space at the end it is stripped from the id in the "ID Numbers" section but apparently is retained in the url linked to the "Read" button. This throws and error rather than taking you to the reader. e.g.:
https://openlibrary.org/books/OL26221263M/Corolla_Sancti_Eadmundi
clicking the "Read Online" button takes you to https://www.archive.org/stream/corollasanctiead00hervuoft%20?ref=ol
Reporter: @JeffKaplan

**Comment by LeadSongDog** [comment]
Can ia simply be trimmed after https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/search/inside.html#L51 ?

**Comment by tfmorris** [comment]
It doesn't make sense to have all downstream consumers have to fix the same problem when it can be fixed at source. The database (and import/update process) should be fixed.

**Comment by hornc** [comment]
I have just bot fixed 206 of these ocaids with spaces, which are all the ones I could find from scanning the ol_dump_editions_2017-04-30.txt data dump file.
See https://openlibrary.org/recentchanges/2017/05/20#bots for the changes. I also normalised any unicode found to NFC, and [fixed some Cyrillic romanisation](https://openlibrary.org/recentchanges/2017/05/20?page=5#bots) 

I _think_ that is all the data fixed. Unless the importer is still adding these, I believe this to be resolved.

**Comment by mekarpeles** [comment]
@hornc mentions this could be addressed in the importbot backend, i.e. https://github.com/internetarchive/openlibrary/blob/5f10771790799d49b17b85ffba682cc2d1a56084/openlibrary/plugins/importapi/code.py

**Comment by hornc** [comment]
Command to find ocaids with spaces in the editions data dump:
`egrep 'ocaid": "[^"]*\s'`

There are some recent user additions which have spaces in the ocaid. I am going to investigate adding field validation on the UI.

**Comment by hornc** [comment]
I have checked for spaces in ocaids again, from the Dec 2017 edition dump. There are 113 ocaids which have spaces anywhere in the string (invalid by definition). unfortunately some are recent additions.

The PR above added validation to the edit page, but the Add page works independently https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/books/add.html , and appears to have a different mechanism, so I'm not sure how reusable the approach is :( I overlooked the two ways to get the data in. It would be nice if this validation occurred in one place!
